test/utf32-test.js

```diff
@@ -0,0 +1,88 @@
+    sampleStr = '<?xml version="1.0" encoding="UTF-8"?>\n<俄语>данные</俄语>';
+
+describe('UTF-32LE codec', function() {
+    it('encodes basic strings correctly', function() {
+        assert.equal(iconv.encode(testStr, 'UTF32-LE').toString('hex'), utf32leBuf.toString('hex'));
+    });
+
+    it('decodes basic buffers correctly', function() {
+        assert.equal(iconv.decode(utf32leBuf, 'ucs4le'), testStr);
+    });
+
+    it('decodes uneven length buffers with no error', function() {
+        assert.equal(iconv.decode(new Buffer([0x61, 0, 0, 0, 0]), 'UTF32-LE'), 'a');
+    });
+
+    it('handles invalid surrogates gracefully', function() {
+        var encoded = iconv.encode(testStr2, 'UTF32-LE');
+        assert.equal(iconv.decode(encoded, 'UTF32-LE'), testStr2);
+    });
+
+    it('handles invalid Unicode codepoints gracefully', function() {
+        assert.equal(iconv.decode(utf32leBufWithInvalidChar, 'utf-32'), testStr + '�');
+    });
+});
+
+describe('UTF-32BE codec', function() {
+    it('encodes basic strings correctly', function() {
+        assert.equal(iconv.encode(testStr, 'UTF32-BE').toString('hex'), utf32beBuf.toString('hex'));
+    });
+
+    it('decodes basic buffers correctly', function() {
+        assert.equal(iconv.decode(utf32beBuf, 'ucs4be'), testStr);
+    });
+
+    it('decodes uneven length buffers with no error', function() {
+        assert.equal(iconv.decode(new Buffer([0, 0, 0, 0x61, 0]), 'UTF32-BE'), 'a');
+    });
+
+    it('handles invalid surrogates gracefully', function() {
+        var encoded = iconv.encode(testStr2, 'UTF32-BE');
+        assert.equal(iconv.decode(encoded, 'UTF32-BE'), testStr2);
+    });
+
+    it('handles invalid Unicode codepoints gracefully', function() {
+        assert.equal(iconv.decode(utf32beBufWithInvalidChar, 'utf-32'), testStr + '�');
+    });
+});
+
+describe('UTF-32 general codec', function() {
+    it('Adds BOM when encoding, defaults to UTF-32LE', function() {
+        assert.equal(iconv.encode(testStr, 'utf-32').toString('hex'), utf32leBOM.toString('hex') + utf32leBuf.toString('hex'));
+    });
+
+    it('Doesn\'t add BOM and uses UTF-32BE when specified', function() {
+        assert.equal(iconv.encode(testStr, 'ucs4', {addBOM: false, defaultEncoding: 'ucs4be'}).toString('hex'), utf32beBuf.toString('hex'));
+    });
+
+    it('Correctly decodes UTF-32LE using BOM', function() {
+        assert.equal(iconv.decode(utf32leBufWithBOM, 'utf-32'), testStr);
+    });
+
+    it('Correctly decodes UTF-32LE without BOM', function() {
+        assert.equal(iconv.decode(iconv.encode(sampleStr, 'utf-32-le'), 'utf-32'), sampleStr);
+    });
+
+    it('Correctly decodes UTF-32BE using BOM', function() {
+        assert.equal(iconv.decode(utf32beBufWithBOM, 'utf-32', { stripBOM: false }), '\uFEFF' + testStr);
+    });
+
+    it('Correctly decodes UTF-32BE without BOM', function() {
+        assert.equal(iconv.decode(iconv.encode(sampleStr, 'utf-32-be'), 'utf-32'), sampleStr);
+    });
+});
```

Comment:
great coverage! Can I suggest one more thing - generate a string containing all codepoints up to 0x10FFFF and convert it to utf32 using iconv-lite and native iconv module, then compare.

Reply:
I'm not sure what you mean by the "native iconv module". Do you mean node-iconv?

And would this be a general unit test, or as a separate speed test?

Reply:
Yes, node-iconv. I think it's a regular unit test, just to check we have the same output.